src/webgpu/api/validation/createView.spec.ts

```diff
@@ -1,311 +1,318 @@
+      reified.baseArrayLayer + reified.arrayLayerCount <= textureLayers;
+    if (reified.dimension !== '2d-array') success &&= reified.arrayLayerCount === 1;
 
+    const texture = t.device.createTexture(textureDescriptor);
     t.expectValidationError(() => {
-      texture.createView(descriptor);
-    }, !_success);
+      texture.createView(view);
+    }, !success);
   });
 
-g.test('creating_cube_map_texture_view')
-  .paramsSimple([
-    { dimension: 'cube', arrayLayerCount: 6, _success: true }, // it is OK to create a cube map texture view with arrayLayerCount == 6
-    // it is an error to create a cube map texture view with arrayLayerCount != 6
-    { dimension: 'cube', arrayLayerCount: 3, _success: false },
-    { dimension: 'cube', arrayLayerCount: 7, _success: false },
-    { dimension: 'cube', arrayLayerCount: 12, _success: false },
-    { dimension: 'cube', _success: false },
-    { dimension: 'cube-array', arrayLayerCount: 12, _success: true }, // it is OK to create a cube map array texture view with arrayLayerCount % 6 == 0
-    // it is an error to create a cube map array texture view with arrayLayerCount % 6 != 0
-    { dimension: 'cube-array', arrayLayerCount: 11, _success: false },
-    { dimension: 'cube-array', arrayLayerCount: 13, _success: false },
-  ] as const)
-  .fn(async t => {
-    const { dimension = '2d-array', arrayLayerCount, _success } = t.params;
+g.test('mip_levels')
+  .desc(
+    `Views must have at least one level, and must be within the level of the base texture.
 
-    const texture = t.createTexture({ arrayLayerCount: 16 });
+  - mipLevelCount=0 at various baseMipLevel values
+  - Cases where baseMipLevel+mipLevelCount goes past the end of the texture
+  - Cases with baseMipLevel or mipLevelCount undefined (compares against reference defaulting impl)
+  `
+  )
+  .params(u =>
+    u
+      .combine('dimension', kTextureDimensions)
+      .beginSubcases()
+      .combineWithParams([
+        { textureLayers: 1, textureLevels: 1 },
+        { textureLayers: kLayers, textureLevels: kLevels },
+      ])
+      .combine('view', [
+        {},
+        { baseMipLevel: 0 },
+        { mipLevelCount: 1 },
+        { baseMipLevel: 0, mipLevelCount: 1 },
+        { baseMipLevel: 1, mipLevelCount: 1 },
+        { baseMipLevel: kLevels - 2, mipLevelCount: 1 },
+        { baseMipLevel: kLevels - 1, mipLevelCount: 1 },
+        { baseMipLevel: 0, mipLevelCount: 2 },
+        { baseMipLevel: 1, mipLevelCount: 2 },
+        { baseMipLevel: kLevels - 2, mipLevelCount: 2 },
+        // mipLevelCount == undefined means to use all remaining levels
+        { mipLevelCount: undefined, baseMipLevel: 0 },
+        { mipLevelCount: undefined, baseMipLevel: 1 },
+        { mipLevelCount: undefined, baseMipLevel: kLevels - 1 },
+        { mipLevelCount: undefined, baseMipLevel: kLevels },
+        // mipLevelCount == 0 means zero mip levels, which is never valid
+        { mipLevelCount: 0, baseMipLevel: 0 },
```

Comment:
Looks like we need to add one condition for `success`: const success = reified.mipLevelCount > 0 && ...?

Reply:
oops, totally missed this. good catch!